compiler/src/iree/compiler/Dialect/Stream/Transforms/PackDispatchOperands.cpp

```diff
@@ -108,13 +108,15 @@ static void updateDispatchOp(IREE::Stream::CmdDispatchOp dispatchOp,
     // i1-i31 -> i32 and i33-i63 -> i64
     // TODO(benvanik): don't extend here but instead pack as we can fit 4 i8's
     // into a single i32 and save 4x our push constant capacity.
-    unsigned bitWidth = operand.getType().getIntOrFloatBitWidth();
-    if (bitWidth < 31) {
-      operand = builder.createOrFold<arith::ExtUIOp>(loc, builder.getI32Type(),
-                                                     operand);
-    } else if (bitWidth > 32 && bitWidth < 64) {
-      operand = builder.createOrFold<arith::ExtUIOp>(loc, builder.getI64Type(),
-                                                     operand);
+    if (!isa<ComplexType>(operand.getType())) {
```

Comment:
We should probably have bitwidth correspond to the ending splat / encoding size. This lets us avoid the complex specific check.

Reply:
> Yep. I multiplied it by 2.

My bad. I misunderstood. Working on fixing this.